core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/HttpAuthOption.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi;
+
+import java.util.function.BiConsumer;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.auth.spi.internal.DefaultHttpAuthOption;
+import software.amazon.awssdk.identity.spi.IdentityProperty;
+import software.amazon.awssdk.utils.builder.SdkBuilder;
+
+/**
+ * An authentication scheme option, composed of the scheme ID and properties for use when resolving the identity and signing
+ * the request.
+ *
+ * This is used in the output from the auth scheme resolver. The resolver returns a list of these, in the order the auth scheme
+ * resolver wishes to use them.
+ *
+ * @see HttpAuthScheme
+ */
+@SdkProtectedApi
+public interface HttpAuthOption {
+
+    /**
+     * Get a new builder for creating a {@link HttpAuthOption}.
+     */
+    static Builder builder(String schemeId) {
+        return new DefaultHttpAuthOption.BuilderImpl(schemeId);
+    }
+
+    /**
+     * Retrieve the scheme ID, a unique identifier for the authentication scheme (aws.auth#sigv4, smithy.api#httpBearerAuth).
+     */
+    String schemeId();
+
+    /**
+     * Retrieve the value of an {@link IdentityProperty}.
+     */
+    <T> T identityProperty(IdentityProperty<T> property);
+
+    /**
+     * Retrieve the value of an {@link SignerProperty}.
+     */
+    <T> T signerProperty(SignerProperty<T> property);
+
+    /**
+     * A method to operate on all {@link IdentityProperty} values of this HttpAuthOption.
+     */
+    void forEachIdentityProperty(IdentityPropertyConsumer consumer);
+
+    /**
+     * A method to operate on all {@link SignerProperty} values of this HttpAuthOption.
+     */
+    <T> void forEachSignerProperty(SignerPropertyConsumer consumer);
+
+    /**
+     * Interface for operating on an {@link IdentityProperty} value.
+     */
+    @FunctionalInterface
+    interface IdentityPropertyConsumer<T> extends BiConsumer<IdentityProperty<T>, T> {
+    }
+
+    /**
+     * Interface for operating on an {@link SignerProperty} value.
+     */
+    @FunctionalInterface
+    interface SignerPropertyConsumer {
+        <T> void accept(SignerProperty<T> propertyKey, T propertyValue);
+    }
+
+    interface Builder extends SdkBuilder<Builder, HttpAuthOption> {
```

Comment:
We could consider adding methods for supplying all properties at once per type, if we think it would be useful for users: `identityProperties(Map<>...)`

Reply:
yeah. thought of doing it when we write the calling code, where the usefulness may get evident.